src/main/java/de/tum/in/www1/artemis/web/rest/hestia/ProgrammingExerciseGitDiffReportResource.java

```diff
@@ -169,21 +170,21 @@ public ResponseEntity<ProgrammingExerciseGitDiffReportDTO> getGitDiffReportForSu
     @GetMapping("programming-exercises/{exerciseId}/commits/{commitHash1}/diff-report/{commitHash2}")
     @EnforceAtLeastStudent
     public ResponseEntity<ProgrammingExerciseGitDiffReportDTO> getGitDiffReportForCommits(@PathVariable long exerciseId, @PathVariable String commitHash1,
-            @PathVariable String commitHash2, @RequestParam(required = false) Long participationId, @RequestParam(required = false) RepositoryType repositoryType)
+            @PathVariable String commitHash2, @RequestParam(required = false) Optional<Long> participationId, @RequestParam Optional<RepositoryType> repositoryType)
```

Comment:
We would still need the required flag in the annotation as `true` is the default value.
https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/bind/annotation/RequestParam.html#required()
```suggestion
            @PathVariable String commitHash2, @RequestParam(required = false) Optional<Long> participationId, @RequestParam(required = false) Optional<RepositoryType> repositoryType)
```

Reply:
I don't think this is the case:

"By default, method parameters that use this annotation are required, but you can specify that a method parameter is optional by setting the @RequestParam annotation’s required flag to false or by declaring the argument with an java.util.Optional wrapper" [source](https://docs.spring.io/spring-framework/reference/web/webmvc/mvc-controller/ann-methods/requestparam.html) 

Also I think some tests would fail then, since Artemis seems to test the endpoints quite extensively.

Reply:
You're right. I over read that one. 👍